doc: matter: Added documentation to Matter west commands (#15619)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/adding_clusters.rst

The ZAP file can be edited using `ZCL Advanced Platform`_ (ZAP tool), a third-party tool that is a generic templating engine for applications and libraries based on Zigbee Cluster Library.

> This tool is provided with the Matter repository in the |NCS|.

Reviewer: This is not true anymore.

> This guide uses the :ref:`ug_matter_gs_tools_matter_west_commands` to run the ZAP tool gui application and generate the output ZAP source files.
> To learn more about Matter west commands and how to use the ZAP tool, see the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool` user guide.

Reviewer: ```suggestion
This guide uses the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool` to install and run the ZAP tool GUI, and generate the data model's C++ source files.
```
conciseness suggestion to link directly to the relevant subsection + "ZAP source files" may not convey that the tool generates c++ sources.
